src/keycloak/chart/templates/_helpers.tpl

```diff
@@ -74,3 +74,24 @@ Create the service DNS name.
 {{- define "keycloak.serviceDnsName" -}}
 {{ include "keycloak.fullname" . }}-headless.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}
 {{- end }}
+
+{{/*
+Check external PostgreSQL connection information. Fails when required values are missing or if PostgreSQL is configured when devMode is enabled.
+*/}}
+
+{{- define "keycloak.postgresql.config" -}}
+{{- if not .Values.devMode -}}
+{{- if .Values.postgresql -}}
+{{ $requiredKeys := list "username" "password" "database" "host" "port" }}
```

Comment:
Just thought of an edge case here - username/password might not be required in IRSA setups. Maybe we should only check the other 3 here?

Reply:
Reviewed and not sure IRSA was supported in the past - opened an issue to track that support https://github.com/defenseunicorns/uds-core/issues/570